app/helpers/explore_api.rb

```
@@ -26,18 +27,87 @@ def bbox_from_value(bbox_value)
         %Q{ST_AsText('#{bbox_value}')}
       end
 
-      def get_table_data(visualization)
-        user_table = UserTable.where(user_id: visualization.user_id, name: visualization.name).first
-        return {} if user_table.nil?
-        {
-          rows: user_table.service.rows_counted,
-          size: user_table.service.table_size,
-          geometry_types: user_table.service.geometry_types
-        }
+      def get_visualizations_table_data(visualizations)
+        return {} if visualizations.blank?
+        data = {}
+        tables_by_user = get_tables_by_user(visualizations)
+        tables_by_user.each do |user_id,tables|
+          begin
+            user = User.find(id: user_id)
+            # INFO If we use the model connection we hit the max number of connections due the pooler so we create a
+            # direct connection with the user database and close it after use
+            conn = PG::Connection.open(:dbname => user.database_name, :host => user.database_host, :user => 'postgres')
+            geometry_types = tables_geometry_types(user_id, tables)
+            data = data.merge(compose_tables_data(conn, user_id, user.database_schema, tables, geometry_types))
+          rescue => e
+            CartoDB.notify_error(
```

This is probably on purpose, but, just to be sure, don't we want to re-raise the error?